static/app/utils/replays/highlightNode.tsx

```diff
@@ -61,56 +63,60 @@ export function removeHighlightedNode(replayer: Replayer, props: RemoveHighlight
+    // mirror will be empty and highlight does not get added because node is null
+    if (
+      !node ||
+      !('getBoundingClientRect' in node) ||
+      !replayer.iframe.contentDocument?.body?.contains(node)
+    ) {
+      return null;
+    }
 
-  const node =
-    'nodeId' in props
-      ? mirror.getNode(props.nodeId)
-      : replayer.iframe.contentDocument?.body.querySelector(props.selector);
-
-  // TODO(replays): There is some sort of race condition here when you "rewind" a replay,
-  // mirror will be empty and highlight does not get added because node is null
-  if (
-    !node ||
-    !('getBoundingClientRect' in node) ||
-    !replayer.iframe.contentDocument?.body?.contains(node)
-  ) {
-    return null;
-  }
-
-  // Create a new canvas with the same dimensions as the iframe. We may need to
-  // revisit this strategy as we create a new canvas for every highlight. See
-  // additional notes in removeHighlight() method.
-  const element = node.nodeType === Node.ELEMENT_NODE ? (node as HTMLElement) : null;
+    // Create a new canvas with the same dimensions as the iframe. We may need to
+    // revisit this strategy as we create a new canvas for every highlight. See
+    // additional notes in removeHighlight() method.
+    const element = node.nodeType === Node.ELEMENT_NODE ? (node as HTMLElement) : null;
 
-  if (!element) {
-    return null;
-  }
+    if (!element) {
+      return null;
+    }
 
-  const canvas = document.createElement('canvas');
-  canvas.width = Number(replayer.iframe.width);
-  canvas.height = Number(replayer.iframe.height);
-  canvas.setAttribute('style', 'position:absolute;');
+    const canvas = document.createElement('canvas');
+    canvas.width = Number(replayer.iframe.width);
+    canvas.height = Number(replayer.iframe.height);
+    canvas.setAttribute('style', 'position:absolute;');
 
-  const boundingClientRect = element.getBoundingClientRect();
-  const drawProps = {
-    annotation: props.annotation ?? '',
-    color: props.color ?? DEFAULT_HIGHLIGHT_COLOR,
-    spotlight: props.spotlight ?? false,
-  };
+    const boundingClientRect = element.getBoundingClientRect();
+    const drawProps = {
+      annotation: props.annotation ?? '',
+      color: props.color ?? DEFAULT_HIGHLIGHT_COLOR,
+      spotlight: props.spotlight ?? false,
+    };
 
-  drawCtx(canvas, boundingClientRect, drawProps);
+    drawCtx(canvas, boundingClientRect, drawProps);
 
-  if ('nodeId' in props) {
-    highlightsByNodeId.set(props.nodeId, {canvas});
-  } else {
-    highlightsBySelector.set(props.selector, {canvas});
-  }
+    if ('nodeIds' in props) {
+      highlightsByNodeIds.set(mirror.getId(node), {canvas});
+    } else {
+      highlightsBySelector.set(props.selector, {canvas});
+    }
 
-  wrapper.insertBefore(canvas, replayer.iframe);
+    wrapper.insertBefore(canvas, replayer.iframe);
 
-  return {
-    canvas,
-  };
+    canvases.push({
+      canvas,
+    });
+  }
+  return canvases;
```

Comment:
We can probably drop this, looks like nothing is using the return value.

Reply:
oh good catch!